lib/src/arithmetic/addend_compressor.dart

```diff
@@ -158,15 +177,116 @@ class CompressTerm implements Comparable<CompressTerm> {
   }
 }
 
+/// Base class for bit-level column compressor function
+abstract class BitCompressor extends Module {
+  /// Input bits to compress
+  @protected
+  late final Logic compressBits;
+
+  /// Input terms to compress
+  late final List<CompressTerm> terms;
+
+  /// The addition results [sum] including carry bit
+  Logic get sum => output('sum');
+
+  /// The carry results [carry].
+  Logic get carry => output('carry');
+
+  late final List<List<double>> _delays;
```

Comment:
do you think it may be useful to expose a `@protected` API for setting these delays rather than private, so that if someone has a technology-specific delay they want to set, they could just extend one of the compressors and change the delay numbers?  this also leads to another question: should there be an argument to generate compressors in `ColumnCompressor` (even the smaller ones) rather than just a flag on whether to use 4:2 or not?

Reply:
The bare minimum is to use 2:2 and 3:2 compressors (half and full-adders).  But you have a point -- people have invented even wider compressors.

Reply:
do you want to include changes to support arbitrary delays and user-defined compressors in this PR?  one consideration is this flag to "use 4:2 compressors" is an API that wouldn't really make sense if we later replace it with a list of compressor types that can be used, for example